### PR TITLE
Update user values configmap if it already exists

### PR DESCRIPTION
**Pull request by rossf7**
## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: apptest.go [hunk 1]
```diff
@@ -407,7 +407,7 @@ func (a *AppSetup) createApps(ctx context.Context, apps []App) error {
 		if app.ValuesYAML != "" {
 			userValuesConfigMap = fmt.Sprintf("%s-user-values", app.Name)
 
-			err := a.createUserValuesConfigMap(ctx, userValuesConfigMap, appCRNamespace, app.ValuesYAML)
+			err := a.ensureUserValuesConfigMap(ctx, userValuesConfigMap, appCRNamespace, app.ValuesYAML)
```
Comment (rossf7) [inline apptest.go:410]: Currently we only create the configmap once. 

I think we should update it if it exists. For update tests and for re-running tests locally.

File: apptest.go [hunk 2]
```diff
@@ -497,15 +495,31 @@ func (a *AppSetup) createUserValuesConfigMap(ctx context.Context, name, namespac
...
+		a.logger.Debugf(ctx, "creating configmap '%s/%s'", namespace, name)
+
+		_, err := a.k8sClient.CoreV1().ConfigMaps(namespace).Create(ctx, configMap, metav1.CreateOptions{})
+		if apierrors.IsAlreadyExists(err) {
+			a.logger.Debugf(ctx, "already created configmap '%s/%s'", namespace, name)
+		} else if err != nil {
+			return microerror.Mask(err)
+		} else {
+			a.logger.Debugf(ctx, "created configmap '%s/%s'", namespace, name)
+		}
```
Comment (tomahawk28) [inline apptest.go:509]:
```suggestion
              }
		a.logger.Debugf(ctx, "created configmap '%s/%s'", namespace, name)
```

File: apptest.go [hunk 3]
```diff
@@ -497,15 +495,31 @@ func (a *AppSetup) createUserValuesConfigMap(ctx context.Context, name, namespac
...
-	if apierrors.IsAlreadyExists(err) {
-		a.logger.Debugf(ctx, "already created configmap '%s/%s'", namespace, name)
-	} else if err != nil {
+	_, err := a.k8sClient.CoreV1().ConfigMaps(namespace).Get(ctx, name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		a.logger.Debugf(ctx, "creating configmap '%s/%s'", namespace, name)
+
+		_, err := a.k8sClient.CoreV1().ConfigMaps(namespace).Create(ctx, configMap, metav1.CreateOptions{})
+		if apierrors.IsAlreadyExists(err) {
+			a.logger.Debugf(ctx, "already created configmap '%s/%s'", namespace, name)
```
Comment (tomahawk28) [inline apptest.go:504]: I think we need to update when configmap is already exists, doesn't it?

File: apptest.go (hunk 2 above)
Comment (rossf7) [reply apptest.go]: Thanks I missed that. 👍

File: apptest.go (hunk 3 above)
Comment (rossf7) [reply apptest.go:504]: Sorry about the late reply. We get the CM first and then match on apierrors.IsNotFound(err) to decide whether we should create or update. 

So I think this is already covered. WDYT?

**Review by tomahawk28 — APPROVED** [review]
LGTM

**Pull request merged**